### PR TITLE
Support requesting translations for multiple integrations in one request

### PR DESCRIPTION
**Pull request by bdraco**
## Proposed change

~~Requires https://github.com/home-assistant/core/pull/71979~~

This fixes an issue where the translations didn't appear right away when there are a large number of discoveries.

Onboarding calls look much cleaner now vs https://github.com/home-assistant/core/pull/71979#issue-1237893247

Screen Shot 2022-05-16 at 20 24 16

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [x] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/data/translation.ts [hunk 1]
```diff
@@ -53,14 +53,16 @@ export const getHassTranslations = async (
   language: string,
   category: TranslationCategory,
   integration?: string,
-  config_flow?: boolean
+  config_flow?: boolean,
+  integrations?: string[]
```
Comment (bdraco) [inline src/data/translation.ts:57]: I wasn't sure if it was better to add a new arg, or replace all the existing usage?

**Comment by bdraco** [comment]
Need to figure out how to memorize that is already has them since opening the flow requests again

**Comment by bdraco** [comment]
After reverting to dev,  that seems like it is an existing issue.  AFAICT, They are unconditionally loaded even if they have been fetched before.

File: src/data/translation.ts (hunk 1 above)
Comment (balloob) [reply src/data/translation.ts:57]: Let's make `integration` be string | string[]

File: src/panels/config/integrations/ha-config-integrations.ts [hunk 2]
```diff
@@ -156,17 +156,20 @@ class HaConfigIntegrations extends SubscribeMixin(LitElement) {
...
           // To render title placeholders
           if (flow.context.title_placeholders) {
-            translationsPromisses.push(
-              this.hass.loadBackendTranslation("config", flow.handler)
-            );
+            integrations.add(flow.handler);
           }
           this._fetchManifest(flow.handler);
         });
-        await Promise.all(translationsPromisses);
```
Comment (balloob) [inline src/panels/config/integrations/ha-config-integrations.ts:169]: One of those things that worked great until we got to scale 😄

**Review by balloob — DISMISSED** [review]

File: src/data/translation.ts (hunk 1 above)
Comment (bdraco) [reply src/data/translation.ts]: Yeah.  Thats much better.. Makes this tiny

**Pull request merged**